addon/adapters/application.js

```diff
@@ -142,18 +141,25 @@ export default Ember.Object.extend(FetchMixin, Evented, {
     @return {Promise}
   */
   createResource(resource) {
-    let url = this.get('url');
-    const json = this.serializer.serialize(resource);
-    return this.fetch(url, {
+    return this.fetch(this.get('url'), {
       method: 'POST',
-      body: JSON.stringify(json)
+      body: JSON.stringify(this.serializer.serialize(resource))
     }).then(function(resp) {
       if (resource.toString().match('JSONAPIResource') === null) {
         return resp;
       } else {
-        resource.set('id', resp.get('id') );
-        let json = resp.getProperties('attributes', 'relationships', 'links', 'meta', 'type', 'isNew', 'id');
-        resource.didUpdateResource(json);
+        resource.set('id', resp.get('id'));
+        resource.didUpdateResource(
+          resp.getProperties(
```

Comment:
Seems cleaner and easier to read. Sometimes I like to be conservative on extra lines, but it readability is more important.